File: problems/Python/59. Spiral Matrix II.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        mat = [[0] * n for _ in range(n)]
        l, r = 0, n - 1
        t, b = 0, n - 1
        num = 1
        
        while l <= r and t <= b:
            for i in range(l, r + 1):
                mat[t][i] = num
                num += 1
            t += 1
            
            for i in range(t, b + 1):
                mat[i][r] = num
                num += 1
            r -= 1
            
            for i in range(r, l - 1, -1):
                mat[b][i] = num
                num += 1
            b -= 1
            
            for i in range(b, t - 1, -1):
                mat[i][l] = num
                num += 1
            l += 1
            
        return mat

File: problems/Java/59. Spiral Matrix II.java
class Solution {
    public int[][] generateMatrix(int n) {
        int[][] mat = new int[n][n];
        int l = 0, r = n - 1;
        int t = 0, b = n - 1;
        int num = 1;

        while (l <= r && t <= b) {
            for (int i = l; i <= r; i++) {
                mat[t][i] = num++;
            }
            t++;

            for (int i = t; i <= b; i++) {
                mat[i][r] = num++;
            }
            r--;

            for (int i = r; i >= l; i--) {
                mat[b][i] = num++;
            }
            b--;

            for (int i = b; i >= t; i--) {
                mat[i][l] = num++;
            }
            l++;
        }
        return mat;
    }
}